pagopa/io-app
# chore: [IA-651] Save EU Covid certificate in a user friendly way

## Short description
The EU Covid certificate is saved to the device gallery with a default name generated by the system. This PR sets a user friendly name to the image before saving it. It also collects all the saved certificates in a gallery folder named "IO".

Resolves #3707.

## List of changes proposed in this pull request
- In `ts/utils/share.ts` an optional `album` parameter is added to the `saveImageToGallery` function.
- In `ts/features/euCovidCert/utils/screenshot.ts` `CaptureOptions` is extended to the new `ScreenshotOptions` type, adding the `filename` and `album` property.
- The `captureScreenshot` function renames the screenshot image before saving it to the gallery (using the new helper functions `savePath` and `rename`).
- Some typos are fixed here and there.
- A unit test has been added to verify that filename and album are correctly passed in the `saveImageToGallery` function. 

## How to test
Open a Covid certificate and save it. The following conditions should be met:
- on Android the saved certificate should be named like "Certificazione verde COVID-19" (it depends on the user's locale);
- on both iOS and Android the saved certificate should be contained in a folder named "IO";
- you can save a certificate (or different certificates) multiple times without getting any error (no name clashes).
